js/preview/src/dev.ts

```diff
@@ -89,8 +90,11 @@ function find_frontend_folders(start_path: string): string[] {
 	dir.forEach((dir) => {
 		const filepath = join(start_path, dir);
 		if (fs.existsSync(filepath)) {
-			if (fs.existsSync(join(filepath, "pyproject.toml")))
-				results.push(filepath);
+			if (fs.existsSync(join(filepath, "pyproject.toml"))) {
+				if (parse(fs.readFileSync(join(filepath, "pyproject.toml"))).project.keywords.indexOf("gradio custom component") > -1) {
```

Comment:
@pngwn Let me know if this is bad lol

Reply:
It's fine but we are doing all of the parsing in python. I removed any JS parsing so we didn't parse the file twice.

Reply:
I think we can handle this in examine.py. We can skip any packages that don't match the tag.